src/Nethermind/Nethermind.Trie/BatchedTrieVisitor.cs

```diff
@@ -369,11 +369,11 @@ static void ThrowUnableToResolve(in SmallTrieVisitContext ctx)
     [StructLayout(LayoutKind.Sequential, Pack = 1)]
     private readonly struct Job
     {
-        public readonly ValueHash256 Key;
+        public readonly Hash256 Key;
```

Comment:
Job need to be as small as possible.

Reply:
This is smaller as `Hash256` is 8 bytes (in struct) whereas `ValueHash256` is 32 bytes?

Reply:
Yes, but the actual data still need to be there. Its not shared with a TrieNode at this point.

Reply:
It is an existing `Hash256` though and the `ValueHash256` comes from one; I haven't allocated any new ones (other than in `GetByHash`); but now it avoids allocating a second `Hash256` via `.ToCommitment()` from the job?

Maybe I'm missing something?

Reply:
Existing or not, the job pool is gonna sum up to 4GB at least. Its better to let the GC take back the existing Hash256 to make space for the job pool in this workload at the expense of CPU which is not the main bottleneck here.